cmd/juju/status/history.go

```diff
@@ -302,3 +332,90 @@ func (c *statusHistoryCommand) writeTabular(writer io.Writer, statuses History)
 	}
 	tw.Flush()
 }
+
+type warningLogger interface {
+	Warningf(format string, args ...interface{})
+}
+
+func (c *statusHistoryCommand) getStatusHistoryClients(ctx context.Context, warningLogger warningLogger) ([]HistoryAPI, bool, error) {
+	if c.clients != nil {
+		return c.clients, true, nil
+	}
+
+	controllerClient, err := getControllerDetailsClient(ctx, c)
+	if err != nil {
+		return nil, false, errors.Annotatef(err, "getting controller addresses")
+	}
+	defer controllerClient.Close()
+
+	// If we're connected to a HA facade that is less that 3, that indicates
+	// that the controller details API is not supported, so we fall back to
+	// using the address of the connected controller only.
+	if controllerClient.BestAPIVersion() < 3 {
```

Comment:
nit: this magic number could be a constant, but not sure if it belongs here or the `api` package.